### PR TITLE
Quicker & quieter intervention access check

### PR DESCRIPTION
**Pull request by OptimusRhine**
* turned off InterventionStrategy logging by default, unless turned on via config setting
* added new `Intervention.quick_access_check(user)` function
  * returns `True` as soon as access is confirmed (or `False` if no access confirmed), rather than making additional checks and/or building a `DisplayDetails` object
  * skips html-generating strategies (they normally just return `True` anyway)
* modified reporting stats function to use new `quick_access_check()`
* added to existing InterventionStrategy-related Intervention tests, to confirm that `Intervention.quick_access_check(user)` acts the same as `Intervention.display_for_user(user).access`

Before these changes, reporting dashboard took 13min to load with truenth stg data.
After these changes, reporting dashboard took ~8min~ (7min with recent changes) to load with same truenth stg data.
In other words, while we should definitely continue to make the check more efficient, this change at least is an incremental improvement.

### CODE REVIEW

**Comment by OptimusRhine** [comment]
@pbugni Sure, that makes sense (and should make things cleaner, if we're talking about just putting the check for `LOG_DEBUG_STRATS` into the `_log()` func, and removing all new `silent` param stuff).

I assumed that the logging served some important purpose, but if it's mostly just noise, definitely seems prudent to have it off by default...

File: portal/models/intervention.py [hunk 1]
```diff
@@ -157,6 +157,46 @@ def display_for_user(self, user):
...
+            if func(intervention=self, user=user):
+                return True
+
+        # 2. check user_intervention for access
+        ui = UserIntervention.query.filter_by(
+            user_id=user.id, intervention_id=self.id).first()
+        if ui and ui.access == 'granted':
+            return True
+
+        # 3. check intervention scope for access
```
Comment (pbugni) [inline portal/models/intervention.py]: @OptimusRhine I *believe* we can run this first for the quick check.  You'd need to review what we do alter in the access strategies - but IIRC that only alters things like the html displayed in the card - doesn't alter access itself?

Comment (OptimusRhine) [reply portal/models/intervention.py]: @pbugni I think you're right, it doesn't look like any of the strategies modify `public_access`.

It also looks like we could reverse the order here entirely (i.e. first check for `public_access`, _then_ check the `UserInterventions`, _then_ check the access strategies), unless I'm missing something. That would definitely speed things up...

Comment (pbugni) [reply portal/models/intervention.py]: I'm sure I thought there was good reason for the original order, but if you feel you've adequately confirmed none of the current strategies would *need* to run before the other checks, lets change it.

Please double check - I'm sure I only chose this more inefficient order as it was necessary at the time.

Comment (OptimusRhine) [reply portal/models/intervention.py]: Well, the order actually still makes since for `display_for_user()`:
* The InterventionStrategy loop needs to happen:
  * ...before the UserIntervention query, because the UserIntervention is returned as part of the `DisplayDetails` object. So if a UI is generated in the IS loop, it needs to happen first.
  * ...no matter what, because of the potential for HTML to be generated, etc
* Given the Strategy loop always needs to happen, the `public_access` check placement doesn't really matter, as long as its there - moving it won't change the runtime of the function.

But none of these are concerns for the quick access check, so I'm going to go ahead and switch that order up, see if it helps things.

Comment (OptimusRhine) [reply portal/models/intervention.py]: (switching it up improved things by ~1min)

**Comment by OptimusRhine** [comment]
@pbugni Only other instance that didn't require the `DisplayDetails` return from `display_for_user()`, was in the QuestionnaireBank model. Updated that instance to use `quick_access_check()` instead.
